### ISSUE TITLE
U4pak supports only zlib pak compression right?

### ISSUE THREAD

**Issue by necros2k7** [panzi/rust-u4pak#10]
subj

**Comment by Xenira** [comment]
Yes, it does

**Comment by necros2k7** [comment]
Any way to add other? unrealpak.exe supports more if I`m not wrong

**Comment by Xenira** [comment]
Sure,, but not my priority rn. I will try to get packing and encrypting pak files up to version 11 first.

If you like to contribute have a look at this for unpacking: https://github.com/panzi/rust-u4pak/blob/c1f6394cfc7004c1947e153345432f607cb63233/src/unpack.rs#L196

**Comment by Xenira** [comment]
I had a look at the common formats and two more formats that should be supported are

- [ ] Gzip
- [ ] LZ4

Keep in mind, that developers can specify custom compression methods. Maybe we should find a way to invoke an external decompressor in order to make this more extensible.

**Comment by necros2k7** [comment]
Maybe merge native UE4 unrealpak.exe into u4pak?

**Comment by Xenira** [comment]
I think this defies the purpose of this project and might cause license issues.
Also would need to include one for each pak version.

**Comment by Xenira** [comment]
And as far as I know Gzip and LZ4 are the ones unrealpak supports. Don't know how custom ones work.

**Comment by necros2k7** [comment]
These utils can be called from within u4pak.exe so it`s user problem to put it in root dir with u4pak
Also I thought UE4 was free engine?

**Comment by Xenira** [comment]
Yes, it could be called from within u4pak, but there are a few problems:
- This would be windows only
- Only one version can be used at a time
- Users would need to install unreal engine
- Different versions from unrealpak expect different arguments
- Yes unreal engine is free, but still has a license. Therefore users need to put unrealpak inside the directory themselves. We can't include it inside the repo
- This tool could not be used as library that easy
- Multi threading might get complicated

Also this would not work with custom compression methods as well.

**Comment by Xenira** [comment]
I am making a lot of progress with getting full support for all versions. I guess this project will soon have full compatibility with all major ue versions.

**Comment by panzi** [comment]
Also then why not use unrealpak directly, if you have it anyway?

**Comment by necros2k7** [comment]
> Also then why not use unrealpak directly, if you have it anyway?

You have to choose right version by hand cause they support differents pak versions.

**Comment by bUbblE1968** [comment]
> Also then why not use unrealpak directly, if you have it anyway?

And usage not obvious
